### PR TITLE
Store results in a SQL database

### PR DESCRIPTION
**Pull request by daltonbohning**
Adds utilities for storing and retrieving test results in a SQL
database.

- New "utils" directory for common code
- init_db.sql
  - sql tables, procedures, functions
- db.cnf
  - sql config
- db.py
  - wrapper for db utilities
- db_util.py
  - utilities for interacting with mariadb
- db_import.py
  - import csv into a table
- db_call.py
  - call sql procedure

Signed-off-by: Dalton Bohning <daltonx.bohning@intel.com>

### CODE REVIEW

File: README_database.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Storing and retrieving results in the database
+
+## Initial Setup
+### Schema
+The initial schema is created with `utils/init_db.sql`. This will create the database, tables, and procedures.
```
Comment (ravalsam) [inline README_database.md:5]: So this will not change anything with the existing DB, right?

**Comment by daltonbohning** [comment]
@sylviachanoiyee @ravalsam This still needs some work, but it is mostly functional (on localhost wolf-26). Please let me know how you feel about the general organization and approach. Something to keep in mind is that in `init_db.sql` all of these tables are created in:
```
CREATE DATABASE frontera_performance COMMENT 'frontera performance metrics';
USE frontera_performance;
```
Which means that if we wanted to do something similar for a different system, we could do:
```
CREATE DATABASE endeavour_performance COMMENT 'endeavour performance metrics';
USE endeavour_performance;
```
And create whichever tables we want there, and then re-use the same python logic for interaction

**Review by ravalsam — APPROVED** [review]

File: README_database.md (hunk 1 above)
Comment (daltonbohning) [reply README_database.md:5]: `utils/init_db.sql` will completely wipe the database (and data) and re-create all tables and procedures. For now, this is useful as the tables are changing frequently. E.g. if a column name changes, then it's more difficult to just re-name the column - we would need a migration script. In the future, we will instead have to individually add columns, etc. as desired, and any fundamental name changes would require some sort of migration script. Also, once permissions are set up properly, we can have a single admin user to make these changes so we don't accidentally wipe the database.
Something else I would like to do in the beginning is archive the imported CSV data, so if we need to restructure we can just re-import all data

Comment (ravalsam) [reply README_database.md:5]: Correct we do not want someone by accident to wipe out the db.
Looks good.

**Pull request merged**